### DISCUSSION TITLE
Remote SSH access from VSCodium

### DISCUSSION THREAD

**Discussion by ghost** [VSCodium/vscodium#693] [Q&A] [answered]
If I understand correctly MS extension "Remote Development" is a proprietary one and unable to use within VSCodium.
Is there an alternative extension or any other way to use this project in a remote development?

I don't want to move back to VSCode due to this reason :/

**Comment by mlopezgva** [comment]
If you're on linux/macosx, you can install sshfs and to the same thing. After installing it, and if you don't want to "leave code", open a terminal and mount wahat ever you wan and, after that, you can access the remote server's file tree not only from vscode, but also with your preferred file manager, open documentation with your pdf viewer, etc.

If you already have configured the remote ssh session in your ~/ssh/config file, it's pretty staightforward:

`ssh remoteserver:/path/to/project mount_dir`

Be advised, git over sshfs or similar is not efficient, so it is preferable to login with ssh and issue de commands "locally".

P.S.: remember to `fusermount -u mount_dir` once you're done. :-) Not critical, just for cleanness

**Reply by femto-code** [reply]
Thanks! I already use SSHFS but I cannot get clangd to work on remote machine. I am using VSCodium on Windows and connect via SSH to a linux server for developing C++. However clangd (for code analysis) does not seem to be configurable to use the remote binary... Do you have any advice / experiences here?

**Comment by gdennie** [comment]
It would be good if VSCodium could auto-magically treat a directory as a file system if it contains a .devcontainer/ sud-directory at its root. Said directory may then have a Dockerfile or docker-compose file and perhaps a .devcontainer.json with which it may then manifest the requisite container, configuration of the VSCodium editor, and terminal defaults... Just saying...

**Comment by guysoft** [comment, marked as answer]
What about this extension?
https://open-vsx.org/extension/jeanp413/open-remote-ssh

**Reply by cabiamdos** [reply]
Hello, I have been unable to install this extension from .vsix (https://github.com/jeanp413/open-remote-ssh/issues/56)

Is there any alternative package?

Thanks

**Reply by femto-code** [reply]
@cabiamdos Try installing from open-vsx.org (default marketplace for VS Codium) instead. Activate the extension in argv.json and it should work. I just tried and could connect to remote SSH folder without problems.

**Reply by Chealer** [reply]
Thanks, but I got an error which does not appear to be tracked before I even got to enter the target… I guess there's a reason why it's version 0.0.49🫤

**Comment by daiyam** [comment]
@guysoft I've just discovered it... It's working right now if you are on linux and without fish as your shell...
I've created a PR to add macOS.
I will look to add support to fish shell.

**Reply by guysoft** [reply]
Says now that mac is supported: macOS 10.14+ (Mojave)

**Reply by StitiFatah** [reply]
>You can connect to a running SSH server on the following platforms.
Supported:
    x86_64 Debian 8+, Ubuntu 16.04+, CentOS / RHEL 7+ Linux.
    ARMv7l (AArch32) Raspbian Stretch/9+ (32-bit).
    ARMv8l (AArch64) Ubuntu 18.04+ (64-bit).
    macOS 10.14+ (Mojave)

Using Archlinux so I can't confirm but maybe it means that the remote server should run one of these OSes ? I'm not sure much people use Raspbian or RHEL as their desktop while it's fairly common for  respectively home and enterprise servers

**Comment by divStar** [comment]
For all the poor souls, who cannot connect to their SSH server using the "Open Remote - SSH" extension by jeanp413 (from VSCodium v1.99.32846 on MacOS): I've spent about an hour figuring out why I *could* connect to one SSH server, that uses a password, but not to another, that uses a private key. Of course I could connect to both using `ssh` in the terminal.

Make sure, that **TCP forwarding is allowed** in the `/etc/ssh/sshd_config` file on the server:
```bash
...
AllowTcpForwarding yes # default setting
...
```
If this setting is set to `no`, the connection from within VSCodium **will** fail (while it might work flawlessly via CLI).
I figured this out thanks to this article: https://doiotyourself.com/posts/2023-05-08-SSH-and-VSCodium.

**Reply by tuanmp** [reply]
If I don’t have permission to edit the remote server `sshd_config`, is there any workaround?

**Reply by daiyam** [reply]
no

**Comment by aergic-dev** [comment]
There's also our Zygos - Remote SSH extension, zero config, works out of the box.
https://github.com/aergic-labs/zygos

**Reply by guysoft** [reply]
Hey, can you explain how it differs from the existing: https://open-vsx.org/extension/jeanp413/open-remote-ssh ?

**Reply by aergic-dev** [reply]
Mostly the zero config part.  It auto fixes argv.json, and has a pretty fancy setup for getting the right reh server download whether it's vscodium, code-oss, kiro, trae, etc.   It also works well with our devcontainers extension so that you can do things like home-machine-> remote-ssh -> devcontainer (where the devcontainer runs on the remote machine).   I'm sure there are other differences, I have not done a full comparison.  The jeanp413 extension is very solid, and the reference point everyone weighs against.  I believe it uses node libraries for ssh, where we directly run openssh.  There's tradeoffs there.  I suspect we might be more compatible with unusual .ssh/config setups, where jeanp413's might have less startup latency given it's not running an additional process.
